File: Java/Exercices/workspaces/solid-lsp-start/src/test/java/zenika/solid/lsp/GeometryTest.java
package zenika.solid.lsp;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class GeometryTest {
    
    @Test public void area_should_be_height_times_width_1() {
        area_should_be_height_times_width(new Rectangle());
    }
    
    @Test public void area_should_be_height_times_width_2() {
        area_should_be_height_times_width(new Square());
    }
    
    private void area_should_be_height_times_width(Rectangle rect) {
        rect.setWidth(5);
        rect.setHeight(4);
        assertEquals(20, rect.getArea());
    }
    
}

File: SOLID/Liskov/GeometryTest.cs
using Xunit;

namespace SOLID.Liskov
{
    public class GeometryTest
    {
        [Fact]
        public void Area_should_be_height_times_width_1()
        {
            Rectangle rect = new Rectangle();
            rect.Width = 5;
            rect.Height = 4;
            Assert.Equal(20, rect.Area);
        }

        [Fact]
        public void Area_should_be_height_times_width_2()
        {
            Square rect = new Square(5);
            Assert.Equal(25, rect.Area);
        }
    }
}